### ISSUE TITLE
CrossEntropyLoss returns NaN on an empty batch instead of the typed error mean gives

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#47]
## Summary

Issue #38 made `mean` over an empty extent a typed error so that an empty last batch cannot poison a loss with `NaN`. `CrossEntropyLoss` does not use `mean`; it hand-rolls the batch average and still returns `NaN`.

## Where (main at 853a224)

```rust
// crates/oxmera-nn/src/loss.rs:39-45
log_p
    .mul(&onehot)?
    .sum(&[])?
    .neg()?
    .mul_scalar(1.0 / logits.dims()[0] as f32)
```

With `n == 0` this is `0 * inf = NaN`.

## Reproduction

```rust
let logits = Tensor::zeros([0, 3]);
let target = Tensor::from_vec_i64(vec![], [0])?;
CrossEntropyLoss.forward(&logits, &target)   // Ok([NaN])
logits.mean(&[])                             // Err(InvalidArgument: extent 0 …)  ← #38
```

`MSELoss` and `BCEWithLogitsLoss` go through `mean(&[])` and return the error.

## Fix

Average with `.mean(&[])` over the per-sample negative log-likelihood (`log_p.mul(&onehot)?.sum(&[1])?.neg()?.mean(&[])`), or check `logits.dims()[0] == 0` up front and return the same error `mean` does. Add the empty-batch case to the loss tests for all three losses so the next loss added here inherits the check.

## Done when

- `CrossEntropyLoss` on `[0, classes]` returns `Err`, never `NaN`.
- A test covers the empty batch for every loss in `oxmera-nn`.
